packages/x-date-pickers/src/AdapterLuxon/index.ts

```diff
@@ -1 +1,70 @@
-export { default as AdapterLuxon } from '@date-io/luxon';
+import { DateTime } from 'luxon';
+import BaseAdapterLuxon from '@date-io/luxon';
+import { MuiFormatTokenMap, MuiPickerFieldAdapter } from '../internals/models';
+import { buildWarning } from '../internals/utils/warning';
+
+const luxonVersionWarning = buildWarning([
+  'Your luxon version does not support `expandFormat`.',
+  'Consider upgrading it to v3.0.2 to have access to the helper text.',
+]);
+
+const formatTokenMap: MuiFormatTokenMap = {
+  // seconds
+  s: 'second',
+  ss: 'second',
+  // minutes
+  m: 'minute',
+  mm: 'minute',
+  // hours
+  H: 'hour',
+  HH: 'hour',
+  h: 'hour',
+  hh: 'hour',
+  // meridiems
+  a: 'am-pm',
+  // dates
+  d: 'day',
+  dd: 'day',
+
+  L: 'month',
+  LL: 'month',
+  LLL: 'month',
+  LLLL: 'month',
+  LLLLL: 'month',
+  // months - format
+  M: 'month',
+  MM: 'month',
+  MMM: 'month',
+  MMMM: 'month',
+  MMMMM: 'month',
+
+  // years
+  y: 'year',
+  yy: 'year',
+  yyyy: 'year',
+};
+
+export class AdapterLuxon extends BaseAdapterLuxon implements MuiPickerFieldAdapter<DateTime> {
+  public formatTokenMap = formatTokenMap;
+
+  // eslint-disable-next-line class-methods-use-this
+  public expandFormat = (format: string) => {
+    if (!DateTime.expandFormat) {
+      throw Error(
+        'Your luxon version does not support `expandFormat`. Consider upgrading it to v3.0.2',
+      );
+    }
+    // The format can contain `yyyyy` which means year between 4 and 6 digits.
```

Comment:
Could you link the source saying that it can contain `yyyyy`, I struggle to understand that one

Reply:
It's not in the doc. It's form [buggy experiments](https://codesandbox.io/s/cool-rain-wkk3bp?file=/index.tsx)

In the codesanbox link above, you can see that `DateTime.expandFormat("f")` return `'M/d/yyyyy, h:m a'` (with 5 `'y'`).

I assume it's for the parsing because formatting keys are limited to 4 `'y'`

In [formatting](https://moment.github.io/luxon/#/formatting?id=table-of-tokens) 
> `yyyy`:  four- to six- digit year, pads to 4 (example: 2014)

In [parsing](https://moment.github.io/luxon/#/parsing?id=table-of-tokens)

> `yyyy`: four-digit year (example: 2014)
> `yyyyy`: four- to six-digit years  (example: 10340)
> `yyyyyy`: six-digit years  (example: 010340)

Reply:
Wow that's ugly

Reply:
I was not sure about how to handle that, And since Hebrew calendars are currently in year 5782 I assumed ignoring 5 and 6 digits would be fine (don't think we will be alive when a 5th digit wi:ml be required 🙀)